csharp/ql/lib/semmle/code/csharp/Attribute.qll

```diff
@@ -99,6 +105,31 @@ class Attribute extends TopLevelExprParent, @attribute {
   override string getAPrimaryQlClass() { result = "Attribute" }
 }
 
+/**
+ * A generic attribute, for example `[...]` on line 1 in
+ *
+ * ```csharp
+ * [MyGenericAttribute<int>(0)]
+ * public void SomeMethod(string s) { }
+ * ```
+ */
+class GenericAttribute extends Attribute {
+  private ConstructedGeneric type;
```

Comment:
Maybe this should be ConstructedType.

Reply:
I think you're right. (It probably doesn't change much as there's only one type (`ConstructedType`) that's both `ConstructedGeneric` and `Type`, which is the return type of `Attribute::getType` used in the char pred)